sympy/functions/special/delta_functions.py

```diff
@@ -191,7 +191,7 @@ def eval(cls, arg, k=0):
         arg = sympify(arg)
         if arg is S.NaN:
             return S.NaN
-        if arg.is_positive or arg.is_negative:
+        if arg.is_positive or arg.is_negative or arg.is_nonzero:
```

Comment:
@sampadsaha5 Thanks for the suggestion!
@Upabjojr  Complex numbers are checked with another if. If _arg_  has imaginary part it returns value error.